pilot/pkg/security/authz/model/model.go

```diff
@@ -287,14 +287,21 @@ func replaceTrustDomainAliases(trustDomain string, trustDomainAliases []string,
 	principalsIncludingAliases := []string{}
 	for _, principal := range principals {
 		trustDomainFromPrincipal := getTrustDomain(principal)
-		//TODO(pitlv2109): Handle * and prefix/suffix.
-		if trustDomainFromPrincipal == "" {
+		// Return the existing principals if getTrustDomain() returns an empty string or "*".
```

Comment:
Before adding more changes, could you first move replaceTrustDomainAliases() and related functions out of the model.go and put them either in util.go or in a different file?

Reply:
I have the change locally. I'll send a PR out and we can get that in first.

Reply:
I sent out a PR here https://github.com/istio/istio/pull/18066. Thank you!